core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -422,7 +426,12 @@ private String getPackageNameFromClassName(String className) {
         }
 
         ClassReader cr = new ClassReader(bytes);
-        ClassWriter writer = new QuarkusClassWriter(cr, ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS);
+        ClassWriter writer = new ClassWriter(cr, ClassWriter.COMPUTE_FRAMES | ClassWriter.COMPUTE_MAXS) {
```

Comment:
In order to avoid creating anonymous classes, it would be nice to introduce a new constructor to `QuarkusClassWriter` that takes a ClassLoader (defaulting to `Thread.currentThread().getContextClassLoader()`)

Reply:
It's not necessarily the same thing, as the TCCL at creation might not match the TCCL at usage. I am not sure if all the current usages would be safe.

Reply:
`return this.cl==null? Thread.currentThread().getContextClassLoader(): cl`?